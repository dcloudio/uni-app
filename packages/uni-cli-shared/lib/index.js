const tags = require('./tags')

const {
  getJson,
  parseJson
} = require('./json')

const {
  getH5Options,
  getManifestJson,
  getNetworkTimeout,
  parseManifestJson
} = require('./manifest.js')

const {
  getMainEntry,
  getNVueMainEntry,
  parseEntry,
  parsePages,
  getPagesJson,
  parsePagesJson
} = require('./pages')

const {
  md5,
  hasOwn,
  hasModule,
  hashify,
  camelize,
  hyphenate,
  removeExt,
  normalizePath,
  getComponentName,
  convertStaticStyle,
  getTemplatePath,
  createSource,
  deleteAsset,
  showRunPrompt
} = require('./util')

const {
  getFlexDirection,
  getPlatformProject,
  isSupportSubPackages,
  getPlatforms,
  getPlatformScss,
  getPlatformSass,
  runByHBuilderX,
  isInHBuilderX,
  isInHBuilderXAlpha,
  getPlatformExts,
  getPlatformTarget,
  getShadowCss,
  getPlatformCssVars,
  getPlatformCssnano,
  getShadowTemplate,
  jsPreprocessOptions,
  cssPreprocessOptions,
  htmlPreprocessOptions,
  nvueJsPreprocessOptions,
  nvueCssPreprocessOptions,
  nvueHtmlPreprocessOptions,
  getPlatformGlobal,
  getPlatformStat,
  getPlatformPush,
  getPlatformUniCloud,
  getDevUniConsoleCode
} = require('./platform')

const uts = require('./uts')

const {
  parseTheme,
  initTheme
} = require('./theme')

module.exports = {
  uts,
  md5,
  tags,
  hasOwn,
  getJson,
  parseJson,
  hashify,
  hasModule,
  camelize,
  hyphenate,
  removeExt,
  normalizePath,
  parseEntry,
  parsePages,
  getH5Options,
  getMainEntry,
  getNVueMainEntry,
  getPagesJson,
  getManifestJson,
  getNetworkTimeout,
  runByHBuilderX,
  isInHBuilderX,
  isInHBuilderXAlpha,
  isSupportSubPackages,
  getPlatforms,
  getFlexDirection,
  getPlatformScss,
  getPlatformSass,
  getPlatformExts,
  getPlatformTarget,
  getPlatformProject,
  getShadowCss,
  getPlatformCssVars,
  getPlatformCssnano,
  getShadowTemplate,
  parsePagesJson,
  parseManifestJson,
  getComponentName,
  convertStaticStyle,
  getTemplatePath,
  createSource,
  deleteAsset,
  showRunPrompt,
  jsPreprocessOptions,
  cssPreprocessOptions,
  htmlPreprocessOptions,
  nvueJsPreprocessOptions,
  nvueCssPreprocessOptions,
  nvueHtmlPreprocessOptions,
  getPlatformGlobal,
  getPlatformStat,
  getPlatformPush,
  getPlatformUniCloud,
  parseTheme,
  initTheme,
  getDevUniConsoleCode
}
