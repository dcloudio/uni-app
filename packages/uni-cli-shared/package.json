{
  "name": "@dcloudio/uni-cli-shared",
  "version": "2.0.0-28920200907001",
  "description": "uni-cli-shared",
  "main": "lib/index.js",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/dcloudio/uni-app.git",
    "directory": "packages/uni-cli-shared"
  },
  "files": [
    "lib",
    "template",
    "components"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "fxy060608",
  "license": "Apache-2.0",
  "dependencies": {
    "hash-sum": "^1.0.2",
    "postcss-urlrewrite": "^0.2.2",
    "strip-json-comments": "^2.0.1"
  },
  "gitHead": "e51484a49748f7bff86bb833763e5dff1522d380"
}
