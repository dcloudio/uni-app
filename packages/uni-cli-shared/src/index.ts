export * from './fs'
export * from './mp'
export * from './url'
export * from './env'
export * from './hbx'
export * from './ssr'
export * from './vue'
export * from './uts'
export * from './logs'
export * from './i18n'
export * from './deps'
export * from './json'
export * from './vite'
export * from './utils'
export * from './easycom'
export * from './constants'
export * from './preprocess'
export * from './postcss'
export * from './filter'
export * from './esbuild'
export * from './resolve'
export * from './scripts'
export * from './platform'
export * from './utsUtils'

export {
  parseUniExtApi,
  parseUniExtApis,
  parseInjects,
  Define,
  DefineOptions,
  Defines,
  getUniExtApiProviderRegisters,
  formatExtApiProviderName,
} from './uni_modules'
export {
  parseUniModulesArtifacts,
  resolveEncryptUniModule,
} from './uni_modules.cloud'
import type { EncryptPackageJson } from './uni_modules.cloud'
export type EncryptArtifacts = EncryptPackageJson['uni_modules']['artifacts']

export { M } from './messages'

export * from './exports'
export { checkUpdate } from './checkUpdate'
