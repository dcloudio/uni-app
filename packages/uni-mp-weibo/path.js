const path = require('path')
module.exports = {
  src: path.join(__dirname, 'src'),
  lib: path.join(__dirname, 'lib')
}
