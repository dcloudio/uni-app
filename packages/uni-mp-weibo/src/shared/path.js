export function parsePath (path) {
  let hash = ''
  let query = ''

  const hashIndex = path.indexOf('#')
  if (hashIndex >= 0) {
    hash = path.slice(hashIndex)
    path = path.slice(0, hashIndex)
  }

  const queryIndex = path.indexOf('?')
  if (queryIndex >= 0) {
    query = path.slice(queryIndex + 1)
    path = path.slice(0, queryIndex)
  }

  return {
    path,
    query,
    hash
  }
}
