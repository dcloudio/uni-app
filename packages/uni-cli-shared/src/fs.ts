import fs from 'fs'
import path from 'path'

export function emptyDir(dir: string, skip: string[] = []) {
  for (const file of fs.readdirSync(dir)) {
    if (skip.includes(file)) {
      continue
    }
    // node >= 14.14.0
    fs.rmSync(path.resolve(dir, file), { recursive: true, force: true })
  }
}
