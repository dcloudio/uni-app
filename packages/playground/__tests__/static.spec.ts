import fs from 'fs-extra'
import path from 'path'
import execa from 'execa'
import { sync } from 'fast-glob'
import { normalizePath } from '@dcloudio/uni-cli-shared'

const projectDir = path.resolve(__dirname, '../static')

describe('static playground', () => {
  jest.setTimeout(50 * 1000)
  const types = {
    'uni-app': [
      // dev 目前面临需要exit的问题
      'dev:app',
      'dev:app-harmony',
      'dev:mp-alipay',
      'dev:mp-baidu',
      'dev:mp-kuaishou',
      'dev:mp-lark',
      'dev:mp-qq',
      'dev:mp-toutiao',
      'dev:mp-weixin',
      'build:app',
      'build:app-harmony',
      'build:h5',
      'build:mp-alipay',
      'build:mp-baidu',
      'build:mp-kuaishou',
      'build:mp-lark',
      'build:mp-qq',
      'build:mp-toutiao',
      'build:mp-weixin',
    ],
    'uni-app-x': [
      'dev:app-android',
      'dev:app-harmony',
      'dev:app-ios',
      'build:app-android',
      'build:app-harmony',
      'build:app-ios',
      'build:h5',
      //   'build:mp-alipay',
      //   'build:mp-baidu',
      //   'build:mp-kuaishou',
      //   'build:mp-lark',
      //   'build:mp-qq',
      //   'build:mp-toutiao',
      //   'build:mp-weixin',
    ],
  }
  const distDir = path.resolve(projectDir, 'dist')
  if (fs.existsSync(distDir)) {
    fs.emptyDirSync(distDir)
  }
  Object.keys(types).forEach((type) => {
    const scripts = types[type]
    scripts.forEach((script) => {
      const mode = script.split(':')[0]
      const platform = script.split(':')[1]
      test(`${type} ${script}`, async () => {
        const outDir = path.resolve(distDir, mode, type, platform)
        console.log(`${type} npm run ${script} start`)
        await execa('npm', ['run', script], {
          cwd: projectDir,
          env: {
            ...process.env,
            UNI_OUTPUT_DIR: outDir,
            UNI_APP_X: type === 'uni-app-x' ? 'true' : 'false',
          },
        })
        console.log(`${type} npm run ${script} end`)
        expect(
          sync('**/test.json', { cwd: outDir, absolute: false })
            .sort()
            .map((filename) => normalizePath(filename))
        ).toMatchSnapshot()
      })
    })
  })
})
