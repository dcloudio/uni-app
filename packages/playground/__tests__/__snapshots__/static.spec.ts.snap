// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`static playground uni-app build:app 1`] = `
[
  "pages/sub/static/app-plus/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-plus/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-plus/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:app-harmony 1`] = `
[
  "pages/sub/static/app-harmony/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-harmony/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-harmony/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:h5 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/static/h5/test.json",
  "pages/sub/static/web/test.json",
  "static/application/test.json",
  "static/h5/test.json",
  "static/web/test.json",
  "uni_modules/test-static/static/application/test.json",
  "uni_modules/test-static/static/h5/test.json",
  "uni_modules/test-static/static/web/test.json",
]
`;

exports[`static playground uni-app build:mp-alipay 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:mp-baidu 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:mp-kuaishou 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:mp-lark 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:mp-qq 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:mp-toutiao 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app build:mp-weixin 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/static/mp-weixin/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "static/mp-weixin/test.json",
  "uni_modules/test-static/static/application/test.json",
  "uni_modules/test-static/static/mp-weixin/test.json",
]
`;

exports[`static playground uni-app dev:app 1`] = `
[
  "pages/sub/static/app-plus/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-plus/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-plus/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:app-harmony 1`] = `
[
  "pages/sub/static/app-harmony/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-harmony/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-harmony/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:mp-alipay 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:mp-baidu 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:mp-kuaishou 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:mp-lark 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:mp-qq 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:mp-toutiao 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app dev:mp-weixin 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/static/mp-weixin/test.json",
  "pages/sub/test/test.json",
  "static/application/test.json",
  "static/mp-weixin/test.json",
  "uni_modules/test-static/static/application/test.json",
  "uni_modules/test-static/static/mp-weixin/test.json",
]
`;

exports[`static playground uni-app-x build:app-android 1`] = `
[
  "pages/sub/static/app-android/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-android/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-android/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app-x build:app-harmony 1`] = `
[
  "pages/sub/static/app-harmony/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-harmony/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-harmony/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app-x build:app-ios 1`] = `
[
  "pages/sub/static/app-ios/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-ios/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-ios/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app-x build:h5 1`] = `
[
  "pages/sub/static/application/test.json",
  "pages/sub/static/h5/test.json",
  "pages/sub/static/web/test.json",
  "static/application/test.json",
  "static/h5/test.json",
  "static/web/test.json",
  "uni_modules/test-static/static/application/test.json",
  "uni_modules/test-static/static/h5/test.json",
  "uni_modules/test-static/static/web/test.json",
]
`;

exports[`static playground uni-app-x dev:app-android 1`] = `
[
  "pages/sub/static/app-android/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-android/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-android/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app-x dev:app-harmony 1`] = `
[
  "pages/sub/static/app-harmony/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-harmony/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-harmony/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;

exports[`static playground uni-app-x dev:app-ios 1`] = `
[
  "pages/sub/static/app-ios/test.json",
  "pages/sub/static/app/test.json",
  "pages/sub/static/application/test.json",
  "static/app-ios/test.json",
  "static/app/test.json",
  "static/application/test.json",
  "uni_modules/test-static/static/app-ios/test.json",
  "uni_modules/test-static/static/app/test.json",
  "uni_modules/test-static/static/application/test.json",
]
`;
